### ISSUE TITLE
Bug: Radio buttons do not get checked with an array of inputs

### ISSUE THREAD

**Issue by vamcs** [facebook/react#18779]
React version: 16.13.0

## Steps To Reproduce

1. Example below illustrates the problem.
2. The component `Radio` has an array of objects that are mapped into `<input>`s which can't be selected. The version without the array in `HelloWorld.jsx` works as expected.

Link to code example: https://glitch.com/~react-input-radio-bug (click on `Remix` below the rendered example to see the code) (if a link to CodeSandbox is really needed I can create one later, please say so)

## The current behavior
The radio buttons do not work or in some cases take two clicks to get marked as checked.

## The expected behavior
It should work as in the version without an array.

**Comment by TodorTotev** [comment]
Hello. The problem is not with React. Your current method returns "on"
![Screenshot 2020-04-29 at 21 49 36](https://user-images.githubusercontent.com/51530311/80635436-8de24380-8a64-11ea-9800-bf293e5e6636.png)

You are passing a method from the main component which you are overriding after that with another method.

Solution to your problem would be, to pass a method from the "HelloWorld" component.

```
  const handleChange = (e) => {
    e.preventDefault();
    setTest(e.target.value);
  };
```

Then pass the method to the Radio component
```
  <Radio value={test} onChange={handleChange} />
```

Then, in the Radio component, declare a value and only pass the method to the onChange event.

```
const Radio = ({ onChange, value }) => {
  return (
    <div style={{ display: "flex" }}>
      {options.map((o) => (
        <label key={o.value}>
          <input
            style={{ width: "auto", display: "inline-block" }}
            type="radio"
            name="language"
            onChange={onChange}
            value={o.value}
            checked={value === o.value}
          />
          {o.label}
        </label>
      ))}
    </div>
  );
};
```

Let me know if you need any more help.

**Comment by vamcs** [comment]
Hi, thanks for taking a look. True, I completely missed setting the value prop for the input, that's why this example doesn't work. By the way, `preventDefault` prevents it from working too. We have some problem with a similar radio button in our codebase and I thought I had managed to reproduce it externally. I'll double-check tomorrow and close this issue if I still can't reproduce it.

**Comment by vamcs** [comment]
I found the problem in my application. Two groups of radio inputs were being rendered with the same name `language` application wide and they conflict, affecting which input was being marked as checked. This forced the user to click twice to make an input being shown as clicked. Also, I've updated the linked example with a way to reproduce it for documentation. To solve that the inputs need to be inside different `<form>`s. Closing the issue.
